rasa/core/policies/rule_policy.py

```diff
@@ -664,12 +663,15 @@ def _is_rule_applicable(
         return bool(
             # rule is shorter than current turn index
             turn_index >= len(reversed_rule_states)
-            # current rule and state turns are empty
-            or (not reversed_rule_states[turn_index] and not conversation_state)
+            # current rule and state turns are conversation starters
```

Comment:
what do you think of this rewrite?
It avoids repeating the index access with `turn_index` and the `get` call to the dictionary

```python
    def _is_rule_applicable(
        self, rule_key: Text, turn_index: int, conversation_state: State
    ) -> bool:
        """Check if rule is satisfied with current state at turn."""

        # turn_index goes back in time
        reversed_rule_states = list(reversed(self._rule_key_to_state(rule_key)))

        if turn_index >= len(reversed_rule_states):
            return True

        current_previous_action = conversation_state.get(PREVIOUS_ACTION)
        rule_previous_action = reversed_rule_states[turn_index].get(PREVIOUS_ACTION)

        if current_previous_action != rule_previous_action:
            return False

        if not current_previous_action:
            # Ignore other states for conversation starts
            return True

        return self._does_rule_match_state(
            reversed_rule_states[turn_index], conversation_state
        )
```

Reply:
Thanks, I've adopted this rewritten form, but I also added some more comments :-)